### ISSUE TITLE
any documentation?

### ISSUE THREAD

**Issue by xiuliren** [sahandKashani/skeletonization#1]
hi, your repo looks very interesting. Which algorithm is this implementation? Does it work with 3D?
any documentation to refer, such as paper or slides?

**Comment by sahandKashani** [comment]
Hi,

It is an algorithm I had to implement for a course. The reference textbook was "Fundamentals of Digital Image Processing [Anil K. Jain]". This specific algorithm was described on pages 382-383. I've attached a screenshot of the book describing the algorithm.

![fundamentals of digital image processing anil k jain - skeleton algorithms pg 382-383](https://user-images.githubusercontent.com/2393156/27203135-b82fbf26-5224-11e7-9bee-95df70c8d4e0.png)

Technically-speaking, I do have a report I had written on the implementation, but I had explicitly removed it from the git repo since my name was on it and I didn't consider it a "reference" I would want directly on my github account. I attached the report below if you want to have a look.

[final.pdf](https://github.com/sahandKashani/skeletonization/files/1079109/final.pdf)
